packages/jsx/src/target/html/components/tabular.tsx

```diff
@@ -0,0 +1,207 @@
+    // }
+    if (row?.attributes?.halign) {
+        vAlign = toVAlign(row.attributes.valign);
+    }
+    // if (cell?.attributes?.halign) {
+    //     vAlign = toVAlign(cell.attributes.valign);
+    // }
+
+    const toHAlign = (s: string | null | undefined) => {
+        if (s == "center") {
+            return "c";
+        }
+        if (s == "right") {
+            return "r";
+        }
+        if (s == "justify") {
+            return "j";
+        }
+        return "l"; // left
+    };
+
+    let hAlign: string = "l";
+    if (tabular.attributes?.halign) {
+        hAlign = toHAlign(tabular.attributes.halign);
+    }
+    if (col?.attributes?.halign) {
+        hAlign = toHAlign(col.attributes.halign);
+    }
+    if (row?.attributes?.halign) {
+        hAlign = toHAlign(row.attributes.halign);
+    }
+    if (cell?.attributes?.halign) {
+        hAlign = toHAlign(cell.attributes.halign);
+    }
+
+    return `lines ${vAlign} ${hAlign} ${bottomLine} ${topLine} ${leftLine} ${rightLine}`;
+};
+
+export const Tabular: ReplacerComponent = function ({ node }) {
+    const state = React.useContext(PretextStateContext);
+    const rows = node.children.filter(
+        (n): n is XastElement => n.type == "element" && n.name == "row"
+    );
+
+    return (
+        <div className="natural-width tabular-box">
+            <table className="tabular">
+                <tbody>
+                    {rows.map((r, r_index) => {
+                        const isHeader = r.attributes?.header == "yes";
+                        const cells = r.children.filter(
+                            (n): n is XastElement =>
+                                n.type == "element" && n.name == "cell"
+                        );
+                        return (
+                            <tr
+                                className={
+                                    isHeader ? "header-horizontal" : undefined
+                                }
+                            >
+                                {cells.map((c, c_index) => {
+                                    if (isHeader) {
+                                        return (
+                                            <th
+                                                className={cellClasses(
+                                                    node,
+                                                    r_index,
+                                                    c_index
+                                                )}
+                                                scope="col"
+                                            >
+                                                {state.processContent(
+                                                    c.children
+                                                )}
+                                            </th>
+                                        );
+                                    } else {
```

Comment:
I'm a fan of leaving the else branch off if it's unneeded. E.g.
```
if (x) {
   return y
}
return z
```

Cuts down on some indentation

Reply:
This is moot as I found a technique to DRY this up: 

```ts
const CellTag = `t${isHeader ? "h" : "d"}` as keyof JSX.IntrinsicElements;
```